pkg/kritis/secrets/pgpkey_test.go

```diff
@@ -0,0 +1,190 @@
+ksT1gGk9dwC4OY4xx/4bfhOOgT8TKb81FmTOOlTE/AgjXdvM5AwJ8B0L5cetqVp5
+Xx812wucnzuPdl+csVc2yWTnHNv7C+j5UWOhm8+5ObVmUmf1i/kSrFhf6cCZTz8m
+xNgW2ckg+8mHQI+jzNjkKINSbZS5vqYHUAIPobsxtRaU6/P429ZGAHxarmU5H7Eb
+KnkQjFeZZzv7/eOuwr6VCF5CYEq1/UiO218Qeyoveuvhsi70Shn/jD3yI8oc0Q2S
+5r/aMgRpzxAc5EYM83eejBDaeXThlsSn0Z01CIbjt6CeqIBnL/FybaHQf2+i7mv+
+5Wv0rfyhnGMi4u0Wnws7xL9Udk6QlPbfQtlVIyCSo51uEraX5ObLoeb59OIYNCjn
+SuzfMzTm8NLfLEkmC69J8kCCd2EwXj0CQQQcF3wwTmo5ItBeLu5P9QtJ+k64fO+a
+jd2vOybF+GGRxyJwqJGHN+aB84O8XNGOAUSKu8GBakiBEz20quh9+WrOTjJYL8k3
+w5ij0hVWkRvQgVfSTRMnuXagqvQwzIvMPhMvZMYHwOgbI0N18tUajvjY7wKeaEWY
+BnW54TekoVvalzrkyoLDyk3BxTTnSTy4TXDkas3YxV3n1i6t7/D+D+R9/d54NoNM
+0KrfAVDKUZXukgLgAA0XgKhlvjPfbQzrFeSPwysZuSp3ss75A2MRCakXs+yXlang
+neZbk/NxItHEeH4S4S90JAQMersvXbHijwIL4xi8Kj9QCHTHWyRd2Cg/yVexr0RH
+XD87M76HPNI9YvB3KL/D/YD2BThWVCqvs15rCUjiH1nMYXMHPFw2LYuSmvBCa9dv
+FUFZHwZMqGEjONaMf+kD2VtrKbbMerpOtWu2YhvV3d9EgETpv9TDmUNts/35uxPy
+Az0c8OL3fGZrdlhzfMDYaE8/FenYiQG2BBgBCgAgFiEEn9FjBK5EYfph7oQLCyGA
+dvrFtawFAlwj05gCGwwACgkQCyGAdvrFtawmcwv8DoIaV4m3uBSOgQBoHKgUVOlP
+Q5zh7AwBNX3N5MAz9dmLqaqVnx/sRKJVWBaTyeF4+I08hWJgxOH7e20pdHKqrvu9
+YWZRiVi+PuGXDdavA5ooGQG1SdVYPQakoZ9i8ZENDQuXwe7GEweqUHSSOUyp490h
+YSPN/O3uRY0xai02H0K5SSbJgA1xOJ8dsaSbQakb/iAfxnUXR9hX4c3rsSjLnRb9
+JupUOE9JU5nvElgiv7VaQRNCgjUAb7wA/M0xaTUtnlMKJ9E12IlF8atab0VKwvsP
+hCl7FWKHjwZs9YKoJuQcahbHXyYU6CRhyCAEMrGk/KWb0AfwjauH97teps2DdWQg
+hm5lwOFqfPKxubbBE2ceXOgsiGU6W2yBuWO9y72mf7rVOGrsGx74fbO9p3ClhM5E
+oIwdSJXeIcaZDsicGM3SDJp/Y4baHRO1wLF8eWLEeobFD2Zw7oZ2xEEBs3Al2Tq8
+de+uIYNCt+A47rHdv4Xn8SlA14h1q8CS7y4lDB9m
+=1siz
+-----END PGP PRIVATE KEY BLOCK-----`
+
+	public = `-----BEGIN PGP PUBLIC KEY BLOCK-----
+Version: BCPG C# v1.6.1.0
+
+mQENBFv+Y7UBCADGf1/XErc1VjjakMwI5kJumfS+FpzFhIq9MsdzoOOD+o+q7Noj
+0r5Y4acp9AGvA0fA7H15JdyT4DEEcNzUNyQySV/Huto0NT5t1L8JLI70/RhF38LA
+kdSR+Q7Uhf/7+6zTB0nfRnc2nfp24PvWSuUlP8NsgUA6WCFY70w6S2iNnv0WsiLU
+XoHpdPm1ke3LABfc5Ujv/4V38WrGb17RP9y3J1TS7TG9tb4ndAIOSEIoxB2njVp1
+B76ZZWD6WbDxAr4V8X5CBEwyDV4mUE3fcouZXeOgFw5WoLIC14Q3YY1zOoS1SOCY
+0joJZTFfoAlWYorbG4XREcSsSodsvEOFjLInABEBAAG0DHRlc3RAcGdwLmNvbYkB
+HAQQAQIABgUCW/5jtQAKCRA2xb4MQTdVtnaNCAChW9mHAE4B4P4dx2ldjSp8GNvq
+nwa7zInyPeuFznvilLFJ/Jc3SnTvx6oGgGPKcLZ1ov2SCisBJhS1CeJ58OSgjmAD
+R/fG69yS1V6JUiRE9x1UIyzuunagcJ8X0QRPkaHSP8YDd4YSqWPNE5kgleHPvGUv
+SrpNTmI5ocu6ONVMrYZtyWbZVVmkMDa8blvBWWin/nXQLp1aWHQMchiL0Dnkr740
+/MEvxEg6ffx6Ew3kP8Y9sk6+7WqGTZzQ3tVq0LnqhP5B6Qd/muo7HXzJnbj+ygpN
+XEnxYQpJtxmO6KZh6r3ZKAUqZOw8BJmqPkN36yzuxBa/BGeKPD3xWsuPkDBT
+=4l+X
+-----END PGP PUBLIC KEY BLOCK-----`
+)
+
+var secrets = []struct {
+	name       string
+	private    string
+	passphrase string
+	public     string
+}{
+	{"no-passphrase", privNoPassphrase, "", public},
+	{"passphrase", privPassphrase, "test passphrase", public},
+}
+
+func TestPgpKey(t *testing.T) {
+	for _, s := range secrets {
+		t.Run(s.name, func(t *testing.T) {
+			key, err := NewPgpKey(s.private, s.passphrase, s.public)
+			if err != nil {
+				t.Error(err)
+			}
+			if key == nil {
```

Comment:
If possible, compare the entire struct. cmp.Equal/cmp.Diff is good for this.

http://go/go-test-comments#compare-full-structures for additional context.